### ISSUE TITLE
What and where to learn to be good at databases and dynamic sites?

### ISSUE THREAD

**Issue by KarlvanMen** [zellwk/ama#1]
Hi, Zell!

I was just neglected from a a front-end developer position I thought I'm fit for. They said that it's because that my knowledge is not up to the level. Hence, I was thinking that you might show the direction I must take. 
I want to learn more about databases and operating them, but the courses I have seen don't seem to be fit for me. They seem... very unspecific and basic. Next to the databases I want to learn more about how to construct dynamic sites.

Can you help me out?

Thanks upfront,
Karl

**Comment by zellwk** [comment]
What position did you apply for, and why specifically were you rejected? 
What was tested? 
What do you want to achieve as a frontend developer?

**Comment by KarlvanMen** [comment]
- "front-end developer position", "because that my knowledge is not up to the level" (no further explanation)
- They asked me to hand in a homework that consisted of:
  
  [] developing a multiple choice test, 
  
  [] making an ER diagram of a database for all tests, which would include the previous test and user information about the results, 
  
  [] and writing some PHP functions to operate the database.
- Because I didn't get the position, I want to learn more about databases, how to construct them efficiently, and best practices in operating them. Besides that I'm also curious on how does one make a dynamic page, for example, an online shop that updates frequently.

**Comment by zellwk** [comment]
Hey dude, 

1) Frontend developers don't deal with databases. It's a good to know, but that's not the frontend role. 
2) PHP = backend already. 
3) I'll fail that test too :) 

The most important question here is: Why did you apply for a frontend position? What makes you want to do frontend?

**Comment by KarlvanMen** [comment]
Ah, okay.. It appears that I've mistaken what is front and back for web :D Thank you for clearing it up. It was positioned as a front-end position. So... I'm not the only one mistaking. Since I knew some bits of it, I went for that.

To answer your questions, I applied to them, because I know what they're doing and I like it. I have some experience (nothing big, but I do). I like to see the outcome of my work instantly. I like to get those small psychological rewards. And frontend is suited for that. I'm actually an electrical engineer, but I like to do something different and currently I'm seeking for some challenges.
